client/state/selectors/is-gutenframe-enabled.js

```diff
@@ -0,0 +1,51 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { isEnabled } from 'config';
+import isVipSite from 'state/selectors/is-vip-site';
+import { getSiteAdminUrl, isJetpackMinimumVersion, isJetpackSite } from 'state/sites/selectors';
+import { isHttps } from 'lib/url';
+import versionCompare from 'lib/version-compare';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import getWordPressVersion from 'state/selectors/get-wordpress-version';
+import isPluginActive from 'state/selectors/is-plugin-active';
+
+export const isGutenframeEnabled = ( state, siteId ) => {
+	if ( ! siteId ) {
+		return false;
+	}
+
+	// We do want Gutenframe flows for Jetpack and Atomic sites.
+	if ( isJetpackSite( state, siteId ) || isSiteAutomatedTransfer( state, siteId ) ) {
+		// But only once they have been updated to WordPress version 5.0 or greater since it will provide Gutenberg
+		// editor by default.
+		const wpVersion = getWordPressVersion( state, siteId );
+		if ( versionCompare( wpVersion, '5.0', '<' ) ) {
+			return false;
+		}
+
+		// But not if they activated the Classic Editor plugin (effectively opting out of Gutenberg).
+		if ( isPluginActive( state, siteId, 'classic-editor' ) ) {
+			return false;
+		}
+
+		// And also only if they have been updated to Jetpack 7.3 or greater since it will provide a way to handle the
+		// frame nonces verification.
+		if ( ! isJetpackMinimumVersion( state, siteId, '7.3-alpha' ) ) {
+			return false;
+		}
+
+		// But only if the site has a SSL cert since the browser cannot embed insecure content in a resource loaded
+		// over a secure HTTPS connection.
+		if ( ! isHttps( getSiteAdminUrl( state, siteId ) ) ) {
```

Comment:
Can we add this to keep the development environment working?

```suggestion
		if ( ! isHttps( getSiteAdminUrl( state, siteId ) ) && 'development' !== process.env.NODE_ENV ) {
```

Reply:
I think we use `CALYPSO_ENV` rather than `NODE_ENV` for specifying the environment.  Anyway, I'd like to try your first suggestion of checking if the protocols mismatch, so we have the same logic running in all the environments.

Reply:
I realized that matching still isn't the best, since a HTTP parent can display a secure iframe (which is how I came to the code suggestion).

Reply:
Check my last commit 80bc620. I changed the code there to embed the iframe if we are either over a HTTP connection or the iframe URL will be loaded over a secure HTTPS connection.

Reply:
Perfecto! 👍